xrpl/models/transactions/transaction.py

```diff
@@ -253,11 +253,30 @@ class Transaction(BaseModel):
     network_id: Optional[int] = None
     """The network id of the transaction."""
 
+    on_behalf_of: Optional[str] = None
+    """The account that the transaction is being sent on behalf of."""
+
+    delegating_seq: Optional[int] = None
+
+    delegating_ticket_seq: Optional[int] = None
+
     def _get_errors(self: Self) -> Dict[str, str]:
         # import must be here to avoid circular dependencies
         from xrpl.wallet.main import Wallet
 
         errors = super()._get_errors()
+        if self.delegating_seq is not None and self.delegating_ticket_seq is not None:
+            errors["Transaction"] = (
+                "Cannot use both delegating_seq and delegating_ticket_seq"
+            )
+        if self.on_behalf_of is not None and (
+            self.delegating_seq is None and self.delegating_ticket_seq is None
+        ):
+            errors["Transaction"] = (
+                "Must provide delegating_seq or delegating_ticket_seq when using"
+                + " on_behalf_of"
+            )
```

Comment:
Can a user just solely provide `on_behalf_of` and not any delegating fields? If not, then it's missing a case for `on_behalf_of` must be set if either `delegating_seq` or `delegating_ticket_seq` are defined:

`self.on_behalf_of is None and (self.delegating_seq is not None or self.delegating_ticket_seq is not None)`